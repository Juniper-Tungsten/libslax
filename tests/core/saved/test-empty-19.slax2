/* Machine Crafted with Care (tm) by slaxWriter */
version 1.1;

match / {
    <top> {
        var $a = {
            call test($a = 1);
        }
        var $b := {
            call test($a = 2);
        }
        var $c = call/call/call;
        var $d = {
            call test($a = 3);
        }
        var $e = call _ call _ call;
        var $f = {
            call _();
        }
        
        <a> $a;
        <b> $b;
        <c> $c;
        <d> $d;
        <e> $e;
        <f> $f;
    }
}

template test ($a = "A") {
    <output> $a;
}

template _ ($a = "A") {
    <output> $a;
}
