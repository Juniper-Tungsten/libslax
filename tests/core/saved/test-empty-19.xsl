<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" version="1.0" extension-element-prefixes="slax-ext">
  <xsl:template match="/">
    <top>
      <xsl:variable name="a">
        <xsl:call-template name="test">
          <xsl:with-param name="a" select="1"/>
        </xsl:call-template>
      </xsl:variable>
      <xsl:variable name="b-temp-1">
        <xsl:call-template name="test">
          <xsl:with-param name="a" select="2"/>
        </xsl:call-template>
      </xsl:variable>
      <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="b" select="slax-ext:node-set($b-temp-1)"/>
      <xsl:variable name="c" select="call/call/call"/>
      <xsl:variable name="d">
        <xsl:call-template name="test">
          <xsl:with-param name="a" select="3"/>
        </xsl:call-template>
      </xsl:variable>
      <xsl:variable name="e" select="concat(call, call, call)"/>
      <xsl:variable name="f">
        <xsl:call-template name="_"/>
      </xsl:variable>
      <a>
        <xsl:value-of select="$a"/>
      </a>
      <b>
        <xsl:value-of select="$b"/>
      </b>
      <c>
        <xsl:value-of select="$c"/>
      </c>
      <d>
        <xsl:value-of select="$d"/>
      </d>
      <e>
        <xsl:value-of select="$e"/>
      </e>
      <f>
        <xsl:value-of select="$f"/>
      </f>
    </top>
  </xsl:template>
  <xsl:template name="test">
    <xsl:param name="a" select="&quot;A&quot;"/>
    <output>
      <xsl:value-of select="$a"/>
    </output>
  </xsl:template>
  <xsl:template name="_">
    <xsl:param name="a" select="&quot;A&quot;"/>
    <output>
      <xsl:value-of select="$a"/>
    </output>
  </xsl:template>
</xsl:stylesheet>
