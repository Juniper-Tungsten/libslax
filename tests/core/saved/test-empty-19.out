<?xml version="1.0"?>
<top>
  <a>1</a>
  <b>2</b>
  <c/>
  <d>3</d>
  <e/>
  <f>A</f>
</top>
